src/app/patient-dashboard/common/formentry/formentry.component.ts

```diff
@@ -359,52 +356,56 @@ export class FormentryComponent implements OnInit, OnDestroy {
   }
 
   public shouldShowPatientReferralsDialog(data: any): void {
-    // if the user does not fill in a field with the id `patientReferral` or `referralLocation`, this dialog should NOT be displayed
-    // conversely, if the user fills in a field with the id `patientReferral` or `referralLocation`, the dialog should be displayed
     this.submittedEncounter = data;
-
-    const referralQuestion = this.form.searchNodeByQuestionId('patientReferral');
+    // Check if referral location question was answered in the form (question id is `referralLocation`)
     const referralLocation = this.form.searchNodeByQuestionId('referralLocation');
-
-    if (referralQuestion.length > 0 && _.isNil(this.programEncounter)) {
-      // show referrals dialog
+    if (referralLocation.length > 0 && _.first(referralLocation).control.value) {
```

Comment:
const referralLocation = this.form.searchNodeByQuestionId('patientReferral');

referralLocation returns a NodeBase array so it will either be empty or it may have values. Either way, referralLocation.length would always work.